sdk/network/Azure.ResourceManager.Network/tests/Tests/FirewallTests.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using Azure.ResourceManager.Network.Models;
+using Azure.ResourceManager.Network.Tests.Helpers;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Network.Tests
+{
+    public class FirewallTests : NetworkServiceClientTestBase
+    {
+        private ResourceGroup _resourceGroup;
+        private VirtualNetwork _network;
+        private PublicIPAddress _publicIPAddress;
+        private string _firewallName;
+
+        private ResourceIdentifier _resourceGroupIdentifier;
+        private ResourceIdentifier _networkIdentifier;
+        private ResourceIdentifier _publicIPAddressIdentifier;
+
+        public FirewallTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [OneTimeSetUp]
+        public async Task GlobalSetUp()
+        {
+            var rgLro = await GlobalClient.DefaultSubscription.GetResourceGroups().CreateOrUpdateAsync(SessionRecording.GenerateAssetName("FirewallRG-"), new ResourceGroupData(Location.WestUS2));
+            ResourceGroup rg = rgLro.Value;
+            _resourceGroupIdentifier = rg.Id;
```

Comment:
We need a [OneTimeTearDown] to remove the resource group, just in case there are duplicated names for resource groups.